optics-core/src/Optics/Internal/Optic/Subtyping.hs

```diff
@@ -26,17 +27,68 @@ class Is k l where
   -- | Witness of the subtyping relationship.
   implies :: (Constraints k p => r) -> (Constraints l p => r)
 
--- | Overlappable instance for a custom type error.
-instance {-# OVERLAPPABLE #-} TypeError ('ShowType k
-                                         ':<>: 'Text " cannot be used as "
-                                         ':<>: 'ShowType l
-                                        ) => Is k l where
-  implies _ = error "unreachable"
-
 -- | Every kind of optic can be used as itself.
 instance Is k k where
   implies r = r
 
+-- | Overlappable instance for a custom type error.
+instance {-# OVERLAPPABLE #-} TypeError
+  ('ShowType k ':<>: 'Text " cannot be used as " ':<>: 'ShowType l
+   ':$$: 'Text "Perhaps you meant to use one of these:"
```

Comment:
Very neat idea. In general I'm a bit cautious about making error messages too long, as GHC error messages are already rather verbose IMHO, but in this case I think it's worth it.

Nitpick: the verb "use" appears in two different senses, first as "using an optic kind at a superclass" then as "using a function to eliminate an optic". How about "apply" in the second case?
```suggestion
   ':$$: 'Text "Perhaps you meant to apply one of these:"
```

Reply:
You're right about double `use`, but apply feels weird to me...

What about simply `Perhaps you meant one of these:` instead? This is what GHC gives you when you misspell things.

```haskell
λ> view_ _Left

<interactive>:35:1: error:
    • Variable not in scope:
        view_ :: Prism (Either a0 b0) (Either c0 b0) a0 c0 -> t
    • Perhaps you meant one of these:
        ‘view’ (imported from Optics.Getter),
        ‘views’ (imported from Optics.Getter)
```

Reply:
Yes, that seems okay to me. With "apply" I was trying to hint that the suggestions were for replacing the function being applied to the optic, but I agree it feels a bit odd.